### PR TITLE
Rewrite change management

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: src/app.js [hunk 1]
```diff
@@ -54,12 +54,50 @@ go.utils = {
             .then(function(result) {
                 contact.extra.rts_id = headteacher_id.toString();
                 contact.extra.rts_emis = emis.toString();
-                contact.name = im.user.answers.reg_first_name;
-                contact.surname = im.user.answers.reg_surname;
+                contact.extra.registration_origin = "";
+                if (contact.name === null) {
```
Comment (imsickofmaps) [inline src/app.js]: are you sure `contact.name` in production comes back as `null` for unedited records? I'm not sure, I'm just asking if you've checked.

File: src/changemanagement.js [hunk 2]
```diff
@@ -9,31 +10,175 @@ go.cm = function() {
...
+                question: $("Please enter the school's EMIS number that you are currently " +
+                            "registered with. This should have 4-6 digits e.g 4351."),
+
+                next: function(content) {
+                    if (go.utils.check_valid_emis(content, array_emis)) {
+                        var emis = parseInt(content, 10);
+                        return go.utils
+                            .cms_get("data/headteacher/?emis__emis=" + emis, im)
+                            .then(function(result) {
+                                var parsed_result = JSON.parse(result.body);
```
Comment (imsickofmaps) [inline src/changemanagement.js:41]: The `cms_get` would be a better place to handle the errors and parse the result I think.

**Comment by imsickofmaps** [comment]
@gsvr first review done. I think a review of error handling and results parsing in `cms_` calls needs discussing.

**Comment by imsickofmaps** [comment]
:+1: with issues handled in #123

**Pull request merged**